app/models/account/gateway.rb

```diff
@@ -18,6 +20,8 @@ def payment_gateway(**options)
     return if payment_gateway_type.blank?
 
     PaymentGateway.implementation(payment_gateway_type, **options).new(payment_gateway_options || {})
+  rescue ArgumentError => exception
+    raise InvalidSettingsError, exception.message
```

Comment:
My question is whether we should raise here? Or should we return something like `InvalidPaymentGateway` object instead? It seems pretty easy to configure an invalid gateway and handling exceptions might be less straightforward than returning an object that can have programmed the desired behavior :thinking:

Reply:
Well, I'm not sure to be honest whether it's "easy" to misconfigure it... I'm not sure in fact how it is possible :grimacing: (and how that "bad" configuration got there in the first place)

Looks like the current implementation returns `nil` if `payment_gateway_type` is blank. In fact, I am not sure what consequences it might have...

To me throwing an exception seems to be more logical and straightforward than returning a `InvalidPaymentGateway`... but it could be done also.

Reply:
@akostadinov So, what do you think?

I am a bit concerned that we may throw this exception in production. But on the other hand, currently an exception would also be thrown, so the current behavior actually wouldn't change :sweat_smile: It would just give us more useful info about the error in BugSnag.

Reply:
I still believe that we should not throw but return an object. Still this fixes a real issue and is a net improvement in behavior so I'm approving. Thank you.